Scripts/EllosBoonSelectorMod.lua

```diff
@@ -609,6 +674,79 @@ function CloseSeedControlScreen( screen, button )
   OnScreenClosed({ Flag = screen.Name })
 end
 
+function IsSecondRoomRewardEligible(requirements, firstRoomReward)
+  if requirements.RequiredUpgradeableGodTraits ~= nil and requirements.RequiredUpgradeableGodTraits >= 1 and firstRoomReward ~= "Boon" then
+    return false
+  end
+  if requirements.RequiredMaxWeaponUpgrades ~= nil and requirements.RequiredMaxWeaponUpgrades < 1 and firstRoomReward == "Hammer" then
+    return false
+  end
+  if requirements.RequiredMinDepth ~= nil and requirements.RequiredMinDepth >= 2 then
+    return false
+  end
+  if requirements.RequiredMinBiomeDepth ~= nil and requirements.RequiredMinBiomeDepth >= 2 then
+    return false
+  end
+  if requirements.RequiredFalseCosmetics ~= nil then
+    return false
+  end
+  return true
```

Comment:
Just out of curiosity, how did you isolate just these cases.  I'm assuming it uses the nightmare method isGameStateEligible in the real code.

Reply:
Yeah, I just looked at all the different rewards in LootData.RunProgress and picked enough of the GameStateRequirements to eliminate all of the invalid values for the first room.

Another approach would be to construct a correct-looking CurrentRun from scratch, and then actually call IsGameStateEligible, but I wasn't quite willing to tackle that yet.